### PR TITLE
create new document with embedded Access Token, add test coverage for CoreDocument

### LINKED ISSUE #912: Create new document with embedded Access token
<This is intended as a guide when thinking about or creating issues for go-centrifuge>

#### Testing
<Describe required tests: Unit, Integration, Testworld or Functional>

#### Acceptance Criteria
<What is the acceptance criteria of this ticket?>

#### Dependencies and affected external components
<Outline any dependencies and components that could be affected by this>

### PR DESCRIPTION
**Pull request by charleenfei**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Closes #912.

### CODE REVIEW

File: documents/coredocument.go [hunk 1]
```diff
@@ -115,6 +117,28 @@ func NewCoreDocumentWithCollaborators(documentPrefix []byte, collaborators Colla
...
+	}
+	collaborators := &CollaboratorsAccess{
+		ReadCollaborators:      []identity.DID{*did[0]},
+		ReadWriteCollaborators: nil,
+	}
+	cd, err := NewCoreDocumentWithCollaborators(documentPrefix, *collaborators)
+	if err != nil {
+		return nil, err
+	}
+	cd, err = cd.AddAccessToken(ctx, params)
```
Comment (vedhavyas) [inline documents/coredocument.go]: You cant Add the token here like this. This will trigger a new Core Document.

Comment (charleenfei) [reply documents/coredocument.go]: AddAccessToken triggers a new version of the core document, not a new core document.

File: documents/coredocument_test.go [hunk 2]
```diff
@@ -8,6 +8,9 @@ import (
 	"os"
 	"testing"
 
+	"github.com/centrifuge/go-centrifuge/protobufs/gen/go/document"
+	"github.com/centrifuge/go-centrifuge/testingutils/config"
+
```
Comment (vedhavyas) [inline documents/coredocument_test.go]: import spacing

Comment (charleenfei) [reply documents/coredocument_test.go]: 👍

**Pull request merged**